### PR TITLE
Fixes for 1000 mb HGT corrupted at RRFS 3KM parallel run.

### LINKED ISSUE #393: 1000 mb heights for the North American RRFS domain is corrupted
After the ufs-weather-model code being run in the LAM parallels was compiled with upp/10.0.9, it was discovered that the 1000 mb heights were corrupted:

132:1297512681:vt=2021092612:1000 mb:60 hour fcst:HGT Geopotential Height [gpm]:
    ndata=15634112:undef=6238409:mean=-2.33694e+15:min=-1.19702e+20:max=1.01562e+14
    grid_template=20:winds(grid):
        polar stereographic grid: (4784 x 3268) input WE:SN output WE:SN res 56
        North pole lat1 -2.500000 lon1 217.500000 latD 60.000000 lonV 260.000000 dx 3170.000000 m dy 3170.000000 m

Sample history files for testing from a RRFS 3 km run can be found on Mars at /gpfs/dell3/ptmp/Eric.Rogers/rrfs3km_forupptest; the contents of this directory are also on HPSS at /NCEPDEV/emc-meso/1year/Eric.Rogers/misc/RRFS3km_upptest.tar

### PR DESCRIPTION
**Pull request by WenMeng-NOAA**
Add fixes for 1000 mb HGT corrupted at RRFS 3KM parallel run.

### CODE REVIEW

File: sorc/ncep_post.fd/MDL2P.f [hunk 1]
```diff
@@ -3846,6 +3847,7 @@ SUBROUTINE MDL2P(iostatusD3D)
                    DO I=1,IM
                    IF(FSL(I,J)<SPVAL)THEN
                      GRID1(I,J) = FSL(I,J)*GI
+                      if(abs(GRID1(I,J))>1000.)print*,"CCC ", GRID1(I,J)
```
Comment (HuiyaChuang-NOAA) [inline sorc/ncep_post.fd/MDL2P.f]: You may want to comment out this line

File: sorc/ncep_post.fd/MDL2P.f [hunk 2]
```diff
@@ -3866,6 +3868,9 @@ SUBROUTINE MDL2P(iostatusD3D)
                      ENDIF
                      Z1000(I,J) = GRID1(I,J)*GI
                      GRID1(I,J) = Z1000(I,J)
+                       if(abs(GRID1(I,J))>1000.)print*,"AAA ", GRID1(I,J)
```
Comment (HuiyaChuang-NOAA) [inline sorc/ncep_post.fd/MDL2P.f]: You may want to comment this line out

**Review by HuiyaChuang-NOAA — APPROVED** [review]
Look good

File: sorc/ncep_post.fd/MDL2P.f (hunk 1 above)
Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/MDL2P.f]: @HuiyaChuang-NOAA Thanks for catching this debugging line.

**Review by EricRogers-NOAA — APPROVED** [review]
Looks good.

**Comment by WenMeng-NOAA** [comment]
The UPP regression tests were conducted on WCOSS-Dell, WCOSS-Cray and Hera. The baseline for hafs is needed to be recreated with this PR merging.

**Pull request merged**